pyro/contrib/gp/kernels/kernel.py

```diff
@@ -52,3 +63,267 @@ def _slice_input(self, X):
             return X.unsqueeze(1)
         else:
             raise ValueError("Input X must be either 1 or 2 dimensional.")
+
+    # def __add__(self, other):
+    #     return self.add(other)
+    #
+    # def __mul__(self, other):
+    #     return self.mul(other)
+    #
+    # def __radd__(self, other):
+    #     return self.add(other)
+    #
+    # def __rmul__(self, other):
+    #     return self.mul(other)
+    #
+    # def add(self, other, name=None):
```

Comment:
I kind of like the method forms, and we're using this pattern for `Distribution.reshape()` and `Distribution.mask()`. @eb8680 would you be ok with `.add()` and `.mul()` methods being the default usage?

@fehiepsi Let's keep commented code out of the codebase, since that leads to lots of false positives when grepping.

Reply:
Yup, I'll remove them or enable them again based on the outcome of this discussion. :)

Reply:
> would you be ok with .add() and .mul() methods being the default usage?

Since you don't mind, it's fine with me as well

Reply:
So I reversed behaviors. :)